src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -446,10 +446,10 @@ void CoreWorkerTest::TestActorFailure(
       ASSERT_EQ(results.size(), 1);
 
       if (results[0]->HasMetadata()) {
-        // Verify if this is the desired error.
-        std::string meta = std::to_string(static_cast<int>(rpc::ErrorType::ACTOR_DIED));
-        ASSERT_TRUE(memcmp(results[0]->GetMetadata()->Data(), meta.data(), meta.size()) ==
-                    0);
+        // Verify if this is one of the desired errors.
+        int error = std::stoi(BufferToString(*results[0]->GetMetadata()));
+        ASSERT_TRUE(error == rpc::ErrorType::ACTOR_DIED ||
+                    error == rpc::ErrorType::OBJECT_UNRECONSTRUCTABLE);
```

Comment:
`OBJECT_UNRECONSTRUCTABLE` should only happen when `is_direct_call == false`.
Can you change this to 
```
if (is_direct_call) {
  ASSERT_TRUE(error == rpc::ErrorType::ACTOR_DIED);
} else {
  ASSERT_TRUE(error == rpc::ErrorType::ACTOR_DIED || error == rpc::ErrorType::OBJECT_UNRECONSTRUCTABLE);
}
```

Reply:
The problem here is currently ObjectID doesn't have direct_call flags, so we're still using `PLASMA` store provider, instead of `LOCAL_PLASMA` store provider, the former would call `FetchOrReconstruct` periodically, and the `OBJECT_UNRECONSTRUCTABLE`  error is set when `NodeManager::HandleTaskReconstruction` tries to resubmit the task but isn't able to find it both locally and from GCS. 

After the ObjectID change is merged, we can use `LOCAL_PLASMA` provider for direct call objects, and then `OBJECT_UNRECONSTRUCTABLE` error should not happen. For now I think this fix is still needed.

@pcmoritz @raulchen

Reply:
I created another PR for this issue using the approach mentioned above.
https://github.com/ray-project/ray/pull/5406